terraform/rancher/project/vault.tf

```diff
@@ -0,0 +1,108 @@
+resource "helm_release" "vault" {
+  count      = var.eureka ? 1 : 0
+  chart      = "vault"
+  name       = "vault-${var.rancher_project_name}"
+  namespace  = rancher2_namespace.this.id
+  repository = "https://helm.releases.hashicorp.com"
+  version    = "0.28.0"
+  values = [<<-EOF
+global:
+  enabled: true
+server:
+  ingress:
+    enabled: false
+  dev:
+    enabled: true
+  ha:
+    enabled: false
+  service:
+    type: ClusterIP
+    port: 8200
+  dataStorage:
+    enabled: true
+  tls:
+    enabled: false
+    auto:
+      enabled: false
+  extraEnvironmentVars:
+    VAULT_DEV_ROOT_TOKEN_ID: "root"
+  resources:
+    requests:
+      memory: "256Mi"
+      cpu: "100m"
+    limits:
+      memory: "1Gi"
+      cpu: "1024m"
+  backup:
+    enabled: false
+  logLevel: "debug"
+  dataStorage:
+    enabled: false
+  auditLog:
+    enabled: false
+  agentInjector:
+    enabled: false
+  metrics:
+    enabled: false
+  unsealConfig:
+    enabled: false
+ui:
+  enabled: true
+  EOF
+  ]
+}
+
+resource "kubernetes_service" "vault-svc" {
+  metadata {
+    name = "vault-${var.rancher_project_name}-np"
+    namespace = rancher2_namespace.this.id
+  }
+  spec {
+    selector = {
+      "app.kubernetes.io/instance" : "vault-${var.rancher_project_name}"
+      "app.kubernetes.io/name" : "vault"
+    }
+    session_affinity = "ClientIP"
+    port {
+      port        = 80
+      target_port = 8200
+    }
+    type = "NodePort"
+  }
+}
+
+
+resource "kubernetes_ingress_v1" "vault-ingress-ui" {
```

Comment:
Why to not use embedded helm values, such as server.ingress:?

Reply:
helm chart is relying on nginx ingress controller, which can potentially cause many problems, just like we had problems with ReportPortal, it going to be the same Sir.